pkg/store/proxy_heap.go

```diff
@@ -618,6 +618,25 @@ func newAsyncRespSet(
 	}
 }
 
+// hasInternalReplicaLabels returns true if any replica label in the series request is not an
+// external label for the given Client.
+func hasInternalReplicaLabels(st Client, req *storepb.SeriesRequest) bool {
```

Comment:
This will always lead to re-sorting if stores have different external labels right? Like `prometheus_replica` on a sidecar and `rule_replica` on a ruler.